back-end/accounts/views.py

```diff
@@ -1,48 +1,115 @@
+from rest_framework_simplejwt.exceptions import TokenError
+from drf_spectacular.utils import (
+    extend_schema,
+    OpenApiParameter,
+    OpenApiTypes,
+    OpenApiResponse,
+)
+from .models import EmailVerificationToken, PasswordResetToken
 from .serializers import (
     EmptySerializer,
-    UserLoginSerializer,
     AuthUserSerializer,
     UserRegisterSerializer,
-    PasswordChangeSerializer,
+    PasswordResetSerializer,
+    ErrorSerializer,
+    SuccessSerializer,
     AccountProfileSerializer,
+    ChangePasswordSerializer,
+    EmailVerificationSerializer,
+    PasswordResetTokenSerializer,
+    RefreshTokenRequestSerializer,
 )
 
 
 User = get_user_model()
 
 
 class AuthViewSet(viewsets.GenericViewSet):
-    permission_classes = [
-        AllowAny,
-    ]
     serializer_class = EmptySerializer
     serializer_classes = {
-        "login": UserLoginSerializer,
+        "login": AuthUserSerializer,
         "register": UserRegisterSerializer,
-        "password_change": PasswordChangeSerializer,
-        "profile": AccountProfileSerializer,
+        "verify_email": EmailVerificationSerializer,
+        "reset_password": PasswordResetSerializer,
+        "send_password_reset_token": PasswordResetTokenSerializer,
+        "change_password": ChangePasswordSerializer,
+        "get_profile": AccountProfileSerializer,
+        "update_profile": AccountProfileSerializer,
+        "refresh_token": RefreshTokenRequestSerializer,
+        "logout": EmptySerializer,
     }
 
-    @action(
-        methods=[
-            "POST",
-        ],
-        detail=False,
+    def get_permissions(self):
+        permission_classes_dict = {
+            "login": [AllowAny],
+            "register": [AllowAny],
+            "verify_email": [AllowAny],
+            "reset_password": [AllowAny],
+            "send_password_reset_token": [AllowAny],
+            "change_password": [IsAuthenticated],
+            "get_profile": [IsAuthenticated],
+            "update_profile": [IsAuthenticated],
+            "refresh_token": [IsAuthenticated],
+            "logout": [IsAuthenticated],
+        }
+        permission_classes = permission_classes_dict.get(self.action, [IsAuthenticated])
+        return [permission() for permission in permission_classes]
+
+    @extend_schema(
+        operation_id="login",
+        request=AuthUserSerializer,
+        responses={
+            "200": SuccessSerializer,
+            "400": ErrorSerializer,
+        },
+        description="User login",
     )
+    @action(methods=["POST"], detail=False)
     def login(self, request):
         serializer = self.get_serializer(data=request.data)
-        serializer.is_valid(raise_exception=True)
-        user = get_and_authenticate_user(**serializer.validated_data)
-        data = AuthUserSerializer(user).data
-        return Response(data=data, status=status.HTTP_200_OK)
+        try:
+            serializer.is_valid(raise_exception=True)
```

Comment:
add logic to validate the token before login

Reply:
to access the login view  a user is not supposed to be authenticated, thus there is not token to be validated... it uses [AllowAny] permission class